pydantic_ai_slim/pydantic_ai/common_tools/exa.py

```diff
@@ -0,0 +1,397 @@
+    )
+
+
+def exa_answer_tool(api_key: str):
+    """Creates an Exa answer tool.
+
+    Args:
+        api_key: The Exa API key.
+
+            You can get one by signing up at [https://dashboard.exa.ai](https://dashboard.exa.ai).
+    """
+    return Tool[Any](  # type: ignore[reportUnknownArgumentType]
+        ExaAnswerTool(client=AsyncExa(api_key=api_key)).__call__,  # type: ignore[reportUnknownArgumentType]
+        name='exa_answer',
+        description='Generates an AI-powered answer to a question with citations from web sources. Returns a comprehensive answer backed by real sources.',
+    )
+
+
+class ExaToolset(FunctionToolset):
+    """A toolset that provides Exa search tools with a shared client.
+
+    This is more efficient than creating individual tools when using multiple
+    Exa tools, as it shares a single API client across all tools.
+
+    Example:
+    ```python
+    from pydantic_ai import Agent
+    from pydantic_ai.common_tools.exa import ExaToolset
+
+    toolset = ExaToolset(api_key='your-api-key')
+    agent = Agent('openai:gpt-4o', toolsets=[toolset])
+    ```
+    """
+
+    def __init__(
+        self,
+        api_key: str,
+        *,
+        num_results: int = 5,
+        max_characters: int | None = None,
+        include_search: bool = True,
+        include_find_similar: bool = True,
+        include_get_contents: bool = True,
+        include_answer: bool = True,
+        id: str | None = None,
+    ):
+        """Creates an Exa toolset with a shared client.
+
+        Args:
+            api_key: The Exa API key.
+
+                You can get one by signing up at [https://dashboard.exa.ai](https://dashboard.exa.ai).
+            num_results: The number of results to return for search and find_similar. Defaults to 5.
+            max_characters: Maximum characters of text content per result. Use this to limit
+                token usage. Defaults to None (no limit).
+            include_search: Whether to include the search tool. Defaults to True.
+            include_find_similar: Whether to include the find_similar tool. Defaults to True.
+            include_get_contents: Whether to include the get_contents tool. Defaults to True.
+            include_answer: Whether to include the answer tool. Defaults to True.
+            id: Optional ID for the toolset, used for durable execution environments.
+        """
+        client = AsyncExa(api_key=api_key)  # type: ignore[reportUnknownVariableType]
+        tools: list[Tool[Any]] = []
+
+        if include_search:
+            tools.append(
+                Tool[Any](  # type: ignore[reportUnknownArgumentType]
+                    ExaSearchTool(client=client, num_results=num_results, max_characters=max_characters).__call__,
+                    name='exa_search',
+                    description='Searches Exa for the given query and returns the results with content. Exa is a neural search engine that finds high-quality, relevant results.',
```

Comment:
Let's dedupe this from the definitions above. Maybe the tool generator functions can accept aclient?